hoodie-common/src/main/java/com/uber/hoodie/common/util/SerializationUtils.java

```diff
@@ -166,6 +71,33 @@ public static void serialize(final Serializable obj, final OutputStream outputSt
     if (objectData == null) {
       throw new IllegalArgumentException("The byte[] must not be null");
     }
-    return deserialize(new ByteArrayInputStream(objectData));
+    return (T) serializerRef.get().deserialize(objectData);
+  }
+
+  private static class KryoSerializerInstance implements Serializable {
+    public static final int KRYO_SERIALIZER_BUFFER_SIZE = 1048576;
```

Comment:
@ovj qq.. does this buffer grow if large objects come along? is this just an intial size? if so, can you rename the variable as well. if not, do we need to expose a parameter for this?

Reply:
Yes this is an initial buffer size. if we try to write larger object then it will auto grow.

Reply:
renamed it.

Reply:
Thanks.. Ready to merge, as long as integ-test passes..